test/integration/clientHelper.ts

```diff
@@ -0,0 +1,76 @@
+import axios from "axios";
+import { HmacSHA1 } from "crypto-js";
+import { MatrixClient, MemoryStorageProvider, PantalaimonClient } from "matrix-bot-sdk";
+import config from "../../src/config";
+
+/**
+ * Register a user using the synapse admin api.
+ * @param username The username to give the user.
+ * @param displayname The displayname to give the user.
+ * @param password The password to use.
+ * @param admin True to make the user an admin, false otherwise.
+ * @returns The response from synapse.
+ */
+export async function registerUser(username: string, displayname: string, password: string, admin: boolean) {
+    let registerUrl = `${config.homeserverUrl}/_synapse/admin/v1/register`
+    let { data } = await axios.get(registerUrl);
+    let nonce = data.nonce!;
+    let mac = HmacSHA1(`${nonce}\0${username}\0${password}\0${admin ? 'admin' : 'notadmin'}`, 'REGISTRATION_SHARED_SECRET');
+    return await axios.post(registerUrl, {
+        nonce,
+        username,
+        displayname,
+        password,
+        admin,
+        mac: mac.toString()
+    })
+}
+
+/**
+ * Register a new test user with a unique username.
+ * @param isAdmin Whether to make the new user an admin.
+ * @returns A string that is the username and password of a new user. 
+ */
+export async function registerNewTestUser(isAdmin: boolean) {
+    let isUserValid = false;
+    let username;
+    do {
+        username = `test-user-${Math.floor(Math.random() * 100000)}`
```

Comment:
Note: It's a good idea to register with a recognizable username. In case this function is somehow compromised, this will make it harder to abuse it silently.